### ISSUE TITLE
Compiling 'screen-capture-recorder.dll'

### ISSUE THREAD

**Issue by MarcFowler** [rdp/screen-capture-recorder-to-video-windows-free#53]
Hey,

I'd like to recompile 'screen-capture-recorder.dll' because I'd like to change the name of the DirectShow device as it appears inside ffmpeg/vlc/etc and because, more importantly, I am concerned that other software which makes use of this same .dll will have set registry keys to control the width/height etc but I want that controlled by my own application.

When I open the project and try to compile, I'm getting an error about missing 'streams.h'. I've tried installing the DirectX SDK and trying again but this hasn't helped.

I've not got much experience in this so any help would be really appreciated!

Thanks!,

**Comment by rdp** [comment]
you need to install the windows [7.1?] SDK then open the "baseclasses"
project in there and compile it locally Debug and Release.  That "might" be
enough....
more details can be seen by looking at the project settings [include paths,
etc.]

**Comment by MarcFowler** [comment]
Thank you very much for the reply, I really appreciate it. Unfortunately this really has me stuck - it's pretty far from my normal! I'm mostly a Mac guy and I cannot seem to compile the baseclasses project for the life of me.

Would it be absurdly cheeky to ask you to compile the screen-capture-recorder.dll for me under a different name?

**Comment by rdp** [comment]
is this for one time use or for distributing to people "out there"?

**Comment by MarcFowler** [comment]
It would be to distribute to people. My main reason for wanting it renamed is to prevent other software which uses the registry keys from overwriting my own app's settings.. I want to only use the 'screen-capture-recorder.dll' part and use ffmpeg to hook into it, but I want it to always be recording full screen so that I can use a `crop` filter to crop if necessary.

**Comment by rdp** [comment]
you just want the dll's then? 32 bit or 64? both?

**Comment by MarcFowler** [comment]
Yes just the .dll's - both would be excellent. And if you could call it something like 'evp-capture', I would be very much indebted to you.

**Comment by rdp** [comment]
you can rename the dll files as desired.
GL!

**Comment by MarcFowler** [comment]
I did try that but it still shows the virtual device in ffmpeg as 'screen-capture-recorder', and it still reads from the same registry locations. What I meant is that my goal is to change how it shows up there to 'evp-capture' and read from different registry keys too. Just renaming doesn't do that unfortunately!

**Comment by rdp** [comment]
Did you remove the previous one first [you're right though on the location
oops]. regsvr32 /u dll_name.dll

**Comment by MarcFowler** [comment]
Yeah I tried it on a computer which had never had this on it previously.

From what I can tell, if you can actually compile it, it's pretty simple to change the name right? Reading through the source it looks like [this line controls what shows inside VLC/ffmpeg](https://github.com/rdp/screen-capture-recorder-to-video-windows-free/blob/7c58710bd822c2c5edd400f705f15decf6786a9d/source_code/setup.cpp#L127), and the registry key location is specified [in DibHelper.cpp](https://github.com/rdp/screen-capture-recorder-to-video-windows-free/blob/46cf792bade7e796d9acff3261c90f93487b3ae6/source_code/DibHelper.cpp).

But I'm stuck on being able to compile the .dll unfortunately!

**Comment by MarcFowler** [comment]
Sorry to bug you again @rdp but do you think you could help me out on this one?

**Comment by rdp** [comment]
It didn't seem immediately clear to me if github posts the attachments or
not.  Anyway sorry for the wait, try these...

**Comment by rdp** [comment]
https://gist.github.com/rdp/fe289186080a25c15903 was the diff [for my own reference]

**Comment by MarcFowler** [comment]
I am so sorry but it doesn't seem to show any kind of attachment or anything like that where I can get the binary. Obviously if I follow the link through to the diff I can see the code changes but how do I access the compiled version to get it?

Also, you're an absolute hero and I really appreciate your help!

**Comment by rdp** [comment]
what's your normal email?

**Comment by MarcFowler** [comment]
It's marc@dfr.me - thanks!
